File: 0304. Range Sum Query 2D - Immutable/solution_naive.cpp
class NumMatrix {
private:
    vector<vector<int>> m;    
public:
    NumMatrix(vector<vector<int>>& matrix) {
        m.assign(matrix.begin(), matrix.end());
    }
    
    int sumRegion(int row1, int col1, int row2, int col2) {
        int sum = 0;
        for (int row_idx = row1; row_idx <= row2; row_idx+=1) {
            for (int col_idx = col1; col_idx <= col2; col_idx+=1) {
                sum += m[row_idx][col_idx];
            }
        }
        return sum;
    }
};

File: 0304. Range Sum Query 2D - Immutable/solution_naive.py
class NumMatrix:

    def __init__(self, matrix: List[List[int]]):
        self.m = deepcopy(matrix)

    def sumRegion(self, row1: int, col1: int, row2: int, col2: int) -> int:
        summ = 0
        for row_id in range(row1, row2+1):
            for col_id in range(col1, col2+1):
                summ += self.m[row_id][col_id]
        return summ
